rdrf/registry/patients/models.py

```diff
@@ -756,19 +758,38 @@ def hard_delete_clinicaldata(self, patient_id):
         for cd in clinicaldata_models:
             cd.delete()
 
-    def set_consent(self, consent_model, answer=True, commit=True):
+    def notify_consent_change(self, consent_value):
+        if self.clinician and not consent_value.answer:
+            logger.info(f"Notify clinician {self.clinician} for change on {consent_value.consent_question}")
+            template_data = {
+                'patient': self,
+                'clinician': self.clinician,
+                'consent_question': consent_value.consent_question,
+                'answer': consent_value.answer,
```

Comment:
We should pass in a summary of the changes to help the email template editor writing a useful notification email if they wish so.
Which question changed, how and if it was the initial save or not.

Reply:
Updated.